pkg/apis/core/validation/validation.go

```diff
@@ -6964,17 +6964,20 @@ func validateMatchLabelKeys(fldPath *field.Path, matchLabelKeys []string, labelS
 		return nil
 	}
 
+	allErrs := field.ErrorList{}
+	if labelSelector == nil {
+		allErrs = append(allErrs, field.Forbidden(fldPath, "must not be specified when labelSelector is not set"))
+		return allErrs
+	}
```

Comment:
nits:

```suggestion
	if labelSelector == nil {
		return field.ErrorList{field.Forbidden(fldPath, "must not be specified when labelSelector is not set")}
	}
```

Reply:
🤔 I prefer to keep it so that it is consistent with the style of the code in that file and also has better readability.